### PR TITLE
Fix perf data inconsistency by prewarming examples benchmarks

### PR DESCRIPTION
**Pull request by t0yv0**
Noticing that aws-ts-s3-folder shows up as FASTER than aws-js-s3-folder, by a large margin, in our perf data. It can't be true and does not repro locally. This is not great. The hypothesis is that since unlike template benchmarks, this setup here does not call `pulumi new`, the issue is that first Pulumi command ends up downloading plugins needed for the test. This skews the data we want to see. 

The fix attempted here is to run a preview before running a full cycle for measurement.

Test run: https://github.com/pulumi/examples/actions/runs/2974913701

### CODE REVIEW

**Comment by t0yv0** [comment]
This makes the job slower, but I think I'm seeing success! I've downloaded the traces from the artifacts in GitHub Actions, and the trace for empty-update on JS and TS are now identical-ish and around 3.5s on total Pulumi Span 🎉

File: misc/test/performance_test.go [hunk 1]
```diff
@@ -177,6 +178,32 @@ func TestManyResources(t *testing.T) {
...
 
+func programTestAsBenchmark(
+	t *testing.T,
+	bench traces.Benchmark,
+	test integration.ProgramTestOptions) {
+
+	// Run preview only to make sure all needed plugins are
+	// downloaded so that these downloads do not skew
+	// measurements.
+	t.Run("prewarm", func(t *testing.T) {
```
Comment (t0yv0) [inline misc/test/performance_test.go:189]: Go question: does this guarantee prewarm will always run before "benchmark"? It seems to work but I was not sure. I did not want to introduce these blocks but ProgramTest does not like running twice in the same testing.T.

**Review by dixler — APPROVED** [review]
LGTM

File: misc/test/performance_test.go (hunk 1 above)
Comment (RobbieMcKinstry) [reply misc/test/performance_test.go:189]: I think only because they are not marked as parallel (the linter in pu/pu might complain about this actually).

Comment (RobbieMcKinstry) [reply misc/test/performance_test.go:189]: > ProgramTest does not like running twice in the same testing.T.

That's strange. Yeah, I think semantically it doesn't make sense to "prewarm" as a subtest. We can merge -- I don't think this code is super delicate, so if it works, it works. It's okay if it's less than perfect.

**Review by RobbieMcKinstry — APPROVED** [review]

File: misc/test/performance_test.go (hunk 1 above)
Comment (t0yv0) [reply misc/test/performance_test.go:189]: It does seem to work on one example, but we can merge and let it run a few days and get data. This is not customer-facing.

**Pull request merged**